File: 2022/day-1/day-1.py
import os
from functools import reduce
import heapq

script_dir = os.path.dirname(os.path.abspath(__file__))
file_path = os.path.join(script_dir, "day-1.txt")

with open(file_path, "r") as file:
    res_heap = []
    curr_group_sum = 0

    line = file.readline()

    while line:
        if line.strip() == "":
            heapq.heappush(res_heap, curr_group_sum)
            curr_group_sum = 0
        else:
            number = int(line)
            curr_group_sum += number

        line = file.readline()

    heapq.heappush(res_heap, curr_group_sum)

    largest = heapq.nlargest(3, res_heap)
    res = sum(largest)

    print(res)

File: 2015/day-1.ts
const { log } = console;

export default function solution(input: string) {
  let res = 0;

  const actionsMap = {
    "(": (val: number) => val + 1,
    ")": (val: number) => val - 1,
  };

  for (const char of input) {
    res = actionsMap[char as "(" | ")"](res);
  }

  log(res);
}
